libraries/app/database_api.cpp

```diff
@@ -310,6 +312,47 @@ database_api_impl::~database_api_impl()
    elog("freeing database api ${x}", ("x",int64_t(this)) );
 }
 
+//////////////////////////////////////////////////////////////////////
+//                                                                  //
+// Market ticker constructor                                        //
+//                                                                  //
+//////////////////////////////////////////////////////////////////////
+market_ticker::market_ticker(const market_ticker_object& mto,
+                             const fc::time_point_sec& now,
+                             const asset_object& asset_base,
+                             const asset_object& asset_quote)
+{
+   time = now;
+   base = asset_base.symbol;
+   quote = asset_quote.symbol;
+   fc::uint128 bv;
+   fc::uint128 qv;
+   price latest_price = asset( mto.latest_base, mto.base ) / asset( mto.latest_quote, mto.quote );
+   if( mto.base != asset_base.id )
+      latest_price = ~latest_price;
+   latest = database_api_impl::price_to_string( latest_price, asset_base, asset_quote );
```

Comment:
Refactor suggestion (may be out of the scope of this ticket): put database_api_impl::price_to_string in some kind of common area. market_ticker should not have to know about database_api_impl, and database_api_impl::price_to_string shouldn't know about graphene::app::price_to_string.